ui/app/utils/classes/stream-logger.js

```diff
@@ -32,7 +32,17 @@ export default EmberObject.extend(AbstractLogger, {
     let buffer = '';
 
     const decoder = new TextDecoder();
-    const reader = yield logFetch(url).then(res => res.body.getReader(), fetchFailure(url));
+    const reader = yield logFetch(url).then(res => {
+      const reader = res.body.getReader();
+      // It's possible that the logger was stopped between the time
+      // polling was started and the log request responded.
+      // If the logger was stopped, the reader needs to be immediately
+      // canceled to prevent an endless request running in the background.
+      if (this.poll.isRunning) {
+        return reader;
+      }
+      reader.cancel();
+    }, fetchFailure(url));
```

Comment:
I noticed you had similar code for this here in `PollLogger`:

https://github.com/hashicorp/nomad/blob/b7a48a99cc5c196f432f3352e5f5e7ee2d63fc89/ui/app/utils/classes/poll-logger.js#L22

Does similar functionality need adding here also?

Reply:
Nice memory! Or sleuthing. Either way :) 

The PollLogger does not need this since the continuous nature of the request polling happens via the task's infinite while loop, which still gets canceled correctly. 

I also tested this in Safari to make double sure.

Reply:
😆 I was trying to find out where you were setting `this.logFetch` and I just grepped for it and came across `PollLogger`

> The PollLogger does not need this since the continuous nature of the request polling happens via the task's infinite while loop, which still gets canceled correctly.

Sweet!